tests/neg/higher-order-quoted-expressions.scala

```diff
@@ -0,0 +1,44 @@
+// Draft of higher order expressions
+// ---------------------------------
+// A higher order expression is a an expression that may contain unbound variables.
+// Unbound variables are listed as arguments of the type lambda of the expression.
+// Splicing can only be done on Expr[_ <: Any].
+// `apply methods on higher order expression provide a way to replace the unbound variables.
+
+// This can be used for HOAS quoted patterns to not return a lambda.
```

Comment:
Add a link to https://en.wikipedia.org/wiki/Higher-order_abstract_syntax ? It helped me understand what was going on in this example :).

Reply:
Added the link to this paragraph in the GPCE paper. There might be some information that is also useful before that.

<img width="506" alt="Screenshot 2022-02-04 at 16 46 41" src="https://user-images.githubusercontent.com/3648029/152558950-60c9d035-0c75-4014-bf7f-9f6370b00d98.png">